client/src/commands/sugar/commands/deploy/collection.ts

```diff
@@ -33,42 +33,43 @@ export const createCollection = async (
 
   const payer = metaplex.identity().publicKey;
 
-  // Create mint account
-  const createMintAccountIx = await SystemProgram.createAccount({
-    fromPubkey: payer,
-    lamports: await metaplex.connection.getMinimumBalanceForRentExemption(
-      MINT_SIZE
-    ),
-    newAccountPubkey: collectionMintPk,
-    programId: TOKEN_PROGRAM_ID,
-    space: MINT_SIZE,
-  });
-
-  // Initialize mint
-  const initMintIx = await createInitializeMintInstruction(
-    collectionMintPk,
-    0,
-    payer,
-    payer
-  );
-
-  const ataPk = await getAssociatedTokenAddress(collectionMintPk, payer);
-
-  // Create associated account
-  const createAtaIx = await createAssociatedTokenAccountInstruction(
-    payer,
-    ataPk,
-    payer,
-    collectionMintPk
-  );
-
-  // Mint
-  const mintToIx = await createMintToInstruction(
-    collectionMintPk,
-    ataPk,
-    payer,
-    1
-  );
+  const [
+    blockhashInfo,
+    [createAtaIx, mintToIx],
+    createMintAccountIx,
+    initMintIx,
+  ] = await Promise.all([
+    // Fetch the latest blockhash to use to specify the transaction lifetime
+    metaplex.connection.getLatestBlockhash(),
+    // Get the [createAta, mintTo] instructions
+    (async () => {
+      const ataPk = await getAssociatedTokenAddress(collectionMintPk, payer);
+      return await Promise.all([
+        // Create associated account
+        createAssociatedTokenAccountInstruction(
+          payer,
+          ataPk,
+          payer,
+          collectionMintPk
+        ),
+        // Mint
+        createMintToInstruction(collectionMintPk, ataPk, payer, 1),
+      ]);
+    })(),
+    // Create mint account
+    (async () =>
+      SystemProgram.createAccount({
+        fromPubkey: payer,
+        lamports: await metaplex.connection.getMinimumBalanceForRentExemption(
+          MINT_SIZE
+        ),
+        newAccountPubkey: collectionMintPk,
+        programId: TOKEN_PROGRAM_ID,
+        space: MINT_SIZE,
+      }))(),
+    // Initialize mint
+    createInitializeMintInstruction(collectionMintPk, 0, payer, payer),
+  ]);
```

Comment:
I don't think these functions are asynchronous anymore. At the time of writing they were using `PublicKey.findProgramAddress` which basically forced entire applications to be written in async code 😅.

Reply:
Oh, we're bringin' 'em back, baby. I'm glad you didn't change these callsites, because the new web3.js is making these async again.

Reason: the `SubtleCrypto` digest method is async.

Reply:
Almost everything in JS seems to be eventually written in async because that one little async call somewhere 😅

Reply:
Reverted this change in https://github.com/solana-playground/solana-playground/pull/164/commits/1a4c7f750908c86e82dc5729a081fafc8e1b5bd2 — others look great